rules/xcodeproj.bzl

```diff
@@ -1150,7 +1152,7 @@ def _xcodeproj_impl(ctx):
         ),
     ]
 
-xcodeproj = rule(
+_xcodeproj = rule(
```

Comment:
I'd suggest moving the implementation of the legacy rule to `rules/internal/legacy_xcodeproj.bzl` and trying to leave a minimal file here at some point. Migrating the name to `_xcodeproj` isn't necessary if you do this now load it as:
```
load(rules/internal/legacy_xcodeproj.bzl, legacy_xcodeproj=xcodeproj)
```

Reply:
Makes sense to me to move to another file, will do!

Reply:
Done!

Reply:
More details here: https://github.com/bazel-ios/rules_ios/pull/443#issuecomment-1105441681